src/workerd/io/trace.c++

```diff
@@ -714,6 +816,763 @@ tracing::Exception::Exception(rpc::Trace::Exception::Reader reader)
   }
 }
 
+namespace {
+kj::Maybe<kj::Array<kj::byte>> readResumeAttachment(const auto& reader) {
+  if (reader.hasAttachment()) {
+    return kj::heapArray<kj::byte>(reader.getAttachment());
+  }
+  return kj::none;
+}
+}  // namespace
+
+tracing::Resume::Resume(kj::Maybe<kj::Array<kj::byte>> attachment)
+    : attachment(kj::mv(attachment)) {}
+
+tracing::Resume::Resume(rpc::Trace::Resume::Reader reader)
+    : attachment(readResumeAttachment(reader)) {}
+
+void tracing::Resume::copyTo(rpc::Trace::Resume::Builder builder) {
+  KJ_IF_SOME(attach, attachment) {
+    builder.setAttachment(attach);
+  }
+}
+
+tracing::Resume tracing::Resume::clone() {
+  return Resume(attachment.map([](auto& attach) { return kj::heapArray<kj::byte>(attach); }));
+}
+
+tracing::Hibernate::Hibernate() {}
+
+tracing::Hibernate::Hibernate(rpc::Trace::Hibernate::Reader reader) {}
+
+void tracing::Hibernate::copyTo(rpc::Trace::Hibernate::Builder builder) {}
+
+tracing::Hibernate tracing::Hibernate::clone() {
+  return Hibernate();
+}
+
+tracing::Attribute::Attribute(kj::String name, Value&& value)
+    : name(kj::mv(name)),
+      value(kj::arr(kj::mv(value))) {}
+
+tracing::Attribute::Attribute(kj::String name, Values&& value)
+    : name(kj::mv(name)),
+      value(kj::mv(value)) {}
+
+namespace {
+kj::Array<tracing::Attribute::Value> readValues(const rpc::Trace::Attribute::Reader& reader) {
+  static auto readValue =
+      [](rpc::Trace::Attribute::Value::Reader reader) -> tracing::Attribute::Value {
+    auto inner = reader.getInner();
+    switch (inner.which()) {
+      case rpc::Trace::Attribute::Value::Inner::TEXT: {
+        return kj::str(inner.getText());
+      }
+      case rpc::Trace::Attribute::Value::Inner::BOOL: {
+        return inner.getBool();
+      }
+      case rpc::Trace::Attribute::Value::Inner::FLOAT: {
+        return inner.getFloat();
```

Comment:
This should be getFloat64 after the type changed to float64.

Reply:
The `getFloat` is based on the name of the field, not the type. The definition is `float @n : Float64`, therefore `inner.getFloat()` is correct.

Reply:
Ok, my mistake, I saw getFloat64 being used internally.